pyvista/core/pointset.py

```diff
@@ -689,15 +690,27 @@ def __init__(
             # one cell per point (point cloud case)
             verts = self._make_vertex_cells(self.n_points)
             n_verts = self.n_points
+        elif verts is not None:
+            verts = np.asarray(verts)
+            validate_array(verts.flatten(), must_be_padded=True)
 
         # here we use CellArray since we must specify deep and n_faces, etc.
         if verts is not None:
             self.verts = CellArray(verts, n_verts, deep)  # type: ignore
         if strips is not None:
+            if not isinstance(strips, CellArray):
+                strips = np.asarray(strips)
+                validate_array(strips.flatten(), must_be_padded=True)
             self.strips = CellArray(strips, n_strips, deep)  # type: ignore
         if faces is not None:
+            if not isinstance(faces, CellArray):
+                faces = np.asarray(faces)
+                validate_array(faces.flatten(), must_be_padded=True)
```

Comment:
Here and elsewhere you're calling `flatten`, which throws out the shape information, no?

Reply:
I call `.flatten` because the checker must be called with 1D array, then it tries to reshape as a 2D array to quickly check if the connectivity array is regular. It sounds like going round in circles but I think that numpy does not copy data and the performance is not impacted. Anyway, it would be safer to call `.view` instead of `.flatten`/`.reshape`.